typescript/cli/src/deploy/core.ts

```diff
@@ -78,21 +89,30 @@ export async function runCoreDeploy({
   artifactsPath?: string;
   outPath: string;
   skipConfirmation: boolean;
+  dryRun: boolean;
 }) {
-  const { customChains, multiProvider, signer } = await getContext({
+  const { customChains, multiProvider } = await getContext({
     chainConfigPath,
     keyConfig: { key },
     skipConfirmation,
+    dryRun,
   });
 
   if (!chains?.length) {
     if (skipConfirmation) throw new Error('No chains provided');
-    chains = await runMultiChainSelectionStep(
-      customChains,
-      'Select chains to connect',
-      true,
-    );
+    chains = await retrieveChainSelection(customChains, dryRun);
   }
+
+  const signer = (await getSigner({
+    keyConfig: { key },
+    skipConfirmation,
+    dryRun,
+  })) as ethers.Signer;
+
+  if (signer) multiProvider.setSharedSigner(signer);
+
+  if (dryRun) await prepareMultiProviderForDryRun(multiProvider, chains);
```

Comment:
still seems like we are forking after setting the signer?
also this function isnt really "dryrun" specific so maybe rename

Reply:
Discussed offline– refactored control flow here and updated naming convention